### PR TITLE
LTS-12 readiness

### PR DESCRIPTION
**Pull request by angerman**
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by angerman** [comment]
Ha :-)
```

stack2nix: haskell.compiler.ghc843 failed to build via nix
--
  | CallStack (from HasCallStack):
  | error, called at src/Stack2nix/Util.hs:79:22 in stack2nix-0.1.3.1-2Inr9CoRFPF1aMzpywCpvx:Stack2nix.Util

```

File: networking/src/Mockable/Production.hs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+-- `MonadMask Production` is missing `generalBracket`.
+{-# OPTIONS_GHC -Wno-missing-methods    #-}
```
Comment (parsonsmatt) [inline networking/src/Mockable/Production.hs]: That seems like an important thing to fix to me. The new `exceptions` library defines everything in terms of `generalBracket`, so it seems to me like this should cause breakage.

File: wallet-new/src/Cardano/Wallet/API/V1/Types.hs [hunk 2]
```diff
@@ -1491,11 +1491,6 @@ data NodeSettings = NodeSettings {
    , setGitRevision    :: !Text
    } deriving (Show, Eq, Generic)
 
--- See note [Version Orphan]
-instance ToSchema Version where
-    declareNamedSchema _ =
-        pure $ NamedSchema (Just "Version") $ mempty
-            & type_ .~ SwaggerString
```
Comment (parsonsmatt) [inline wallet-new/src/Cardano/Wallet/API/V1/Types.hs]: We can delete [the note, too](https://github.com/input-output-hk/cardano-sl/blob/d8df53edd0ad81bfeb08ddad2c1dbeab1e2ba7d8/wallet-new/src/Cardano/Wallet/API/V1/Types.hs#L1495-L1498) :)

Comment (angerman) [reply wallet-new/src/Cardano/Wallet/API/V1/Types.hs]: Good catch!

File: networking/src/Mockable/Production.hs (hunk 1 above)
Comment (angerman) [reply networking/src/Mockable/Production.hs]: Right! If someone who is more familiar with the code knows how the proper generalBracket implementation has to look can just provide it, that would be helpful :-)

Comment (parsonsmatt) [reply networking/src/Mockable/Production.hs]: `Production` is a newtype around `IO` so we can just use `GeneralizedNewtypeDeriving` to derive the instance.

**Pull request closed without merging**